extensions/ql-vscode/src/language-context-store.ts

```diff
@@ -0,0 +1,40 @@
+import { App } from "./common/app";
+import { DisposableObject } from "./common/disposable-object";
+import { AppEvent, AppEventEmitter } from "./common/events";
+import { QueryLanguage } from "./common/query-language";
+
+type LanguageFilter = QueryLanguage | "All";
+
+export class LanguageContextStore extends DisposableObject {
+  public readonly onLanguageContextChanged: AppEvent<void>;
+  private readonly onLanguageContextChangedEmitter: AppEventEmitter<void>;
+
+  private state: LanguageFilter;
+
+  constructor(app: App) {
+    super();
+    // State initialization
+    this.state = "All";
+
+    // Set up event emitters
+    this.onLanguageContextChangedEmitter = this.push(
+      app.createEventEmitter<void>(),
+    );
+    this.onLanguageContextChanged = this.onLanguageContextChangedEmitter.event;
+  }
+
+  public clearLanguageContext() {
+    this.state = "All";
+    this.onLanguageContextChangedEmitter.fire();
+  }
+
+  public setLanguageContext(language: QueryLanguage) {
+    this.state = language;
+    this.onLanguageContextChangedEmitter.fire();
+  }
+
+  // TODO: comment on why string is used here
```

Comment:
A leftover "TODO" 👀 Do you want to add a comment here, or shall we remove this? 😄

Reply:
I was leaving it to highlight the question, but I did that in the PR message, so maybe I can just remove it :)